README.md

```diff
@@ -30,18 +30,108 @@ Input JSON file with road OpenStreetMap tags:
+    {"type": "travel_lane", "direction": "backward"},
     {"type": "travel_lane", "direction": "forward"},
     {"type": "travel_lane", "direction": "forward"},
     {"type": "sidewalk", "direction": "forward"}
 ]
 ```
 
+Lane Definition
+---------------
+
+It is important to determine a strict definition for a lane.
+An OpenStreetMap way, usually a `highway=*`, can in some cases be divided into lanes,
+where each lane extends from the start to the end of the way.
+A lane's direction of travel is not necessarily the same as the direction of the way.
+A lane represents a continuous area with no formal separation of vehicles.
+Lanes are often separated by something visible like painted lines or raised curbs,
+but these are not necessary for lanes to exist.
+
+
+### Example 1 ###
+
+A road with 5 marked lanes.
+
+```json
+[
+    {"type": "travel_lane", "direction": "backward"},
+    {"type": "travel_lane", "direction": "backward"},
+    {"type": "travel_lane", "direction": "forward"},
+    {"type": "travel_lane", "direction": "forward"},
+    {"type": "travel_lane", "direction": "forward"},
+]
+```
+
+![Example 1](https://upload.wikimedia.org/wikipedia/commons/thumb/0/0e/Gr%C3%BCnbergstra%C3%9Fe_2.JPG/240px-Gr%C3%BCnbergstra%C3%9Fe_2.JPG)
+
+Note: there is no sidewalk or verge for pedestrians to walk here.
+
+### Example 2 ###
+
+A road with 2 marked lanes.
+
+```json
+[
+    {"type": "shoulder", "direction": "both"},
+    {"type": "travel_lane", "direction": "forward"},
+    {"type": "travel_lane", "direction": "backward"},
+    {"type": "shoulder", "direction": "both"},
+]
+```
+
+![Example 2](https://upload.wikimedia.org/wikipedia/commons/f/f0/A537_Cat_and_Fiddle_Road_-_geograph.org.uk_-_175899.jpg)
+
+Note:
+
+- there is are shoulders where pedestrians could walk in both directions.
+- this is in the UK, so traffic drives on the left.
+- the direction is the predominant direction,
+  overtaking vehicles could travel in the opposite direction.
+
+### Example 3 ###
+
+A narrow road with one lane.
+
+```json
+[
+    {"type": "shoulder", "direction": "both"},
+    {"type": "travel_lane", "direction": "both"},
+    {"type": "shoulder", "direction": "both"},
+]
+```
+
+![Example 3](https://upload.wikimedia.org/wikipedia/commons/5/58/Back_Road_In_Ireland.jpg)
+
+TODO: shoulders?
+
+### Discussion Example ###
+
+A road with no marked lanes.
+
+![No Lanes India](https://upload.wikimedia.org/wikipedia/commons/thumb/5/5a/Bijupara-Khalari_Road_-_Jharkhand_1648.JPG/1920px-Bijupara-Khalari_Road_-_Jharkhand_1648.JPG)
+
+![No Lanes Delhi](https://upload.wikimedia.org/wikipedia/commons/thumb/a/a8/Ratan_Lal_Market%2C_Kaseru_Walan%2C_Paharganj%2C_New_Delhi%2C_Delhi%2C_India_-_panoramio_%281%29.jpg/1280px-Ratan_Lal_Market%2C_Kaseru_Walan%2C_Paharganj%2C_New_Delhi%2C_Delhi%2C_India_-_panoramio_%281%29.jpg)
+
+![No Lanes Dutch](https://upload.wikimedia.org/wikipedia/commons/e/e8/Fietsstraat.jpg)
+
+Option 1: Mark the entire road as a single lane in both directions
```

Comment:
This gets my vote -- it's like the narrow lane example above, except not so narrow. Even on the narrow lane, two opposing vehicles would _probably_ stick to the usual driving direction by default, depending on the exact situation and geometry.

For renderers, one big unmarked bidi lane seems easy to handle. Likewise for routing. Things like traffic simulators have to make some choice about how much detail to model, and internally, it'd be fine for them to treat this as two one-direction lanes with conventional directions. I think that modeling choice really will vary, and this library shouldn't get opinionated about it

Reply:
agreed, I will select option 1 but leave the considered options there so it is recorded...